topics/variant-analysis/tutorials/baculovirus-isolate-variation/tutorial.md

```diff
@@ -0,0 +1,867 @@
+> >         print "Error: Required columns (POS, ALLELE, ISOLATE, REL.ALT.0.05) are missing." > "/dev/stderr";
+> >         exit 1;
+> >     }
+> > 
+> >     # Print the header and add the new SPEC column
+> >     print $0, "SPEC";
+> >     next;
+> > }
+> > 
+> > {
+> >     # When reaching a new position, prepare the specificity
+> >     if ($(pos_col) != current_pos) {
+> >         # Assign the calculated specificity to all rows of the current position
+> >         for (i in pos_lines) {
+> >             # Add "SNV specificity: " before the specificity value
+> >             final_spec = (specificity == "" ? "0" : "SNV specificity: " specificity);
+> >             print pos_lines[i], final_spec;
+> >         }
+> >         # Reset variables for the new position
+> >         delete pos_lines;
+> >         specificity = "";
+> >         current_pos = $(pos_col);
+> >     }
+> > 
+> >     # Save the current row for later
+> >     pos_lines[NR] = $0;
+> > 
+> >     # Conditions for calculating specificity
+> >     if ($(allele_col) == "ALT1" && 
+> >         ($(isolate_col) == "CpGV-E2" || $(isolate_col) == "CpGV-S" || $(isolate_col) == "CpGV-M") && 
+> >         $(rel_alt_col) > 0) {
+> >         # Concatenate isolate names with " + " if REL.ALT.0.05 > 0
+> >         specificity = (specificity == "" ? $(isolate_col) : specificity " + " $(isolate_col));
+> >     }
+> > }
+> > 
+> > END {
+> >     # Assign the specificity to the last position
+> >     for (i in pos_lines) {
+> >         # Add "SNV specificity: " before the specificity value
+> >         final_spec = (specificity == "" ? "0" : "SNV specificity: " specificity);
+> >         print pos_lines[i], final_spec;
+> >     }
+> > }
+> > ```
+> {: .code-in}
+>
+>    - Click Run Tool
+> 
+>
+{: .hands_on}
+
+# Visualization of Variable SNV Positions by Specificity
+
+Now, the VCF table has an additional column called `SPEC`, which indicates the SNV specificity for each position. This allows us to create a special SNV plot for each isolate. The special thing about this is that a separate plot is created for each isolate and each specificity. Since we can only look at each isolate individually, we first have to filter out the data for one isolate, we start with CpGV-S. After that, a plot is created for each SNV specificity for the CpGV-S isolate.  
+
+> <hands-on-title> Extract data for one isolate only </hands-on-title>
+>
+> 1. {% tool [Filter data on any column using simple expressions](Filter1) %} with the following parameters:
+>    - {% icon param-file %} *"Filter"*: the VCF table with determined SNV specificities; output of **Text reformatting with awk** {% icon tool %}
+>    - *"With following condition"*: `c25=='CpGV-S'`
+>    - *"Number of header lines to skip"*: `1`
+>    - Click Run Tool
+>
+>    > <comment-title> Filtering the CpGV-S data </comment-title>
+>    >
+>    > The VCF SNV table is reduced to the data of CpGV-S. This is done by filtering on column 23 (ISOLATE).
```

Comment:
Just noticed that this comment talks about column 23, but you're configuring c25 in the hands-on.